kotest-extensions/kotest-extensions-htmlreporter/src/jvmMain/kotlin/io.kotest.extensions.htmlreporter/HtmlReporter.kt

```diff
@@ -0,0 +1,199 @@
+package io.kotest.extensions.htmlreporter
+
+import io.kotest.core.listeners.ProjectListener
+import org.jdom2.DocType
+import org.jdom2.Document
+import org.jdom2.Element
+import org.jdom2.Text
+import org.jdom2.input.SAXBuilder
+import org.jdom2.output.Format
+import org.jdom2.output.XMLOutputter
+import java.io.File
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.nio.file.StandardOpenOption
+
+class HtmlReporter(
```

Comment:
The project listener that locates the test data, and the class that generates HTML could be separated since they are different things. The former could invoke the latter.

Reply:
I don't mind that. What would we call these? `HtmlReporter` and `HtmlGenerator`?

Reply:
HtmlWriter maybe for the thing that does the html generator and HtmlReporter yeah for the listener class.